### PR TITLE
refactor: add Tasklist to current Zeebe distro

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description
Make tasklist packaged in the common distribution

Failing Zeebe integration tests should be fixed by https://github.com/camunda/zeebe/pull/17860
## Related issues

closes #

### CODE REVIEW

**Review by vsgoulart — APPROVED** [review]
From the frontend side everything seems ok, I tested it locally and nothing changed

**Review by romansmirnov — COMMENTED** [review]
Not critical; when starting tasklist it prints Camunda Operate to the console

```
PS C:\Develop\Projects\camunda-cloud\zeebe\dist\target\camunda-zeebe-8.6.0-SNAPSHOT\bin> .\tasklist
   ___   _   __  __ _   _ _  _ ___   _      ___  ___ ___ ___    _ _____ ___
  / __| /_\ |  \/  | | | | \| |   \ /_\    / _ \| _ \ __| _ \  /_\_   _| __|
 | (__ / _ \| |\/| | |_| | .` | |) / _ \  | (_) |  _/ _||   / / _ \| | | _|
  \___/_/ \_\_|  |_|\___/|_|\_|___/_/ \_\  \___/|_| |___|_|_\/_/ \_\_| |___|

                                                           8.6.0-SNAPSHOT


2024-04-29 11:07:36.432 [] [background-preinit] [] INFO
      org.hibernate.validator.internal.util.Version - HV000001: Hibernate Validator 8.0.1.Final
2024-04-29 11:07:36.693 [] [main] [] INFO
```

**Comment by houssain-barouni** [comment]
> Not critical; when starting tasklist it prints Camunda Operate to the console
> 
> ```
> PS C:\Develop\Projects\camunda-cloud\zeebe\dist\target\camunda-zeebe-8.6.0-SNAPSHOT\bin> .\tasklist
>    ___   _   __  __ _   _ _  _ ___   _      ___  ___ ___ ___    _ _____ ___
>   / __| /_\ |  \/  | | | | \| |   \ /_\    / _ \| _ \ __| _ \  /_\_   _| __|
>  | (__ / _ \| |\/| | |_| | .` | |) / _ \  | (_) |  _/ _||   / / _ \| | | _|
>   \___/_/ \_\_|  |_|\___/|_|\_|___/_/ \_\  \___/|_| |___|_|_\/_/ \_\_| |___|
> 
>                                                            8.6.0-SNAPSHOT
> 
> 
> 2024-04-29 11:07:36.432 [] [background-preinit] [] INFO
>       org.hibernate.validator.internal.util.Version - HV000001: Hibernate Validator 8.0.1.Final
> 2024-04-29 11:07:36.693 [] [main] [] INFO
> ```

I will fix it

**Comment by Zelldon** [comment]
Can we maybe just get rid of these banners completely? 🤔 As they also breaking the logging in SaaS anyway 😅

File: dist/pom.xml [hunk 1]
```diff
@@ -124,6 +124,17 @@
...
     </dependency>
 
+    <dependency>
+      <groupId>io.camunda</groupId>
+      <artifactId>tasklist-webapp</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>commons-logging</groupId>
+          <artifactId>commons-logging</artifactId>
+        </exclusion>
```
Comment (romansmirnov) [inline dist/pom.xml:134]:
```suggestion
        </exclusion>
        <exclusion>
          <groupId>org.springframework</groupId>
          <artifactId>spring-webflux</artifactId>
        </exclusion>
```

With graphql, Tasklist includes spring-webflux transitively. Would be great to not include this one in the distribution.

Comment (romansmirnov) [reply dist/pom.xml:134]: Just did a small test locally, and it didn't cause any issues with excluding

**Review by romansmirnov — APPROVED** [review]
LGTM! Great step! Just had the two comments. Besides that, I have tested the distribution locally.

File: dist/src/main/config/log4j2.xml [hunk 2]
```diff
@@ -6,9 +6,9 @@
     <Property name="log.pattern">%d{yyyy-MM-dd HH:mm:ss.SSS} [%X{actor-scheduler}] [%t] [%X{actor-name}] %-5level
       %logger{36} - %msg%n
     </Property>
-    <Property name="log.stackdriver.serviceName">${env:ZEEBE_LOG_STACKDRIVER_SERVICENAME:-${env:OPERATE_LOG_STACKDRIVER_SERVICENAME:-}}
+    <Property name="log.stackdriver.serviceName">${env:ZEEBE_LOG_STACKDRIVER_SERVICENAME:-${env:OPERATE_LOG_STACKDRIVER_SERVICENAME:-${env:TASKLIST_LOG_STACKDRIVER_SERVICENAME:-}}}
```
Comment (Zelldon) [inline dist/src/main/config/log4j2.xml:9]: @romansmirnov @houssain-barouni I guess we have to fix here again the permission issues with creating the logs folder. For zeebe/operate we create the folder in the dockerfile

File: tasklist.Dockerfile [hunk 3]
```diff
@@ -58,12 +58,23 @@ EXPOSE 8080
 RUN apk update && apk upgrade && \
     apk add --no-cache bash openjdk21-jre tzdata
 
-WORKDIR /usr/local/tasklist
+ENV TASKLIST_HOME=/usr/local/tasklist
+
+WORKDIR ${TASKLIST_HOME}
 VOLUME /tmp
+VOLUME ${TASKLIST_HOME}/logs
```
Comment (Zelldon) [inline tasklist.Dockerfile:65]: 👍🏼

File: tasklist/qa/util/src/main/java/io/camunda/tasklist/qa/util/TestContainerUtil.java [hunk 4]
```diff
@@ -398,11 +398,11 @@ public GenericContainer createTasklistContainer(
       String dockerImageName, String version, TestContext testContext) {
     tasklistContainer =
         new GenericContainer<>(String.format("%s:%s", dockerImageName, version))
-            .withExposedPorts(8080)
+            .withExposedPorts(8080, 9600)
```
Comment (Zelldon) [inline tasklist/qa/util/src/main/java/io/camunda/tasklist/qa/util/TestContainerUtil.java:401]: @romansmirnov we need to do the same changes in the charts and docs again, as we did for Operate I guess?

**Review by ChrisKujawa — APPROVED** [review]
Looks good from checking the code/changes, I haven't tested it.

File: tasklist/qa/util/src/main/java/io/camunda/tasklist/qa/util/TestContainerUtil.java (hunk 4 above)
Comment (romansmirnov) [reply tasklist/qa/util/src/main/java/io/camunda/tasklist/qa/util/TestContainerUtil.java:401]: Yes, Houssain is aware of the required changes in the helm charts.

File: dist/pom.xml (hunk 1 above)
Comment (houssain-barouni) [reply dist/pom.xml:134]: I would better exclude it form `com.graphql-java-kickstart:graphql-spring-boot-autoconfigure` in dependency management

**Comment by houssain-barouni** [comment]
> Can we maybe just get rid of these banners completely? 🤔 As they also breaking the logging in SaaS anyway 😅

Let's keep it for now, some people will ask "where is my gorgeous banner?"

File: dist/pom.xml (hunk 1 above)
Comment (houssain-barouni) [reply dist/pom.xml:134]: Fixed here https://github.com/camunda/zeebe/pull/17649/commits/69aff526c8125352b9617d9b2b34930466ae1311

**Pull request merged**